Common/Securities/Future/FuturesExpiryUtilityFunctions.cs

```diff
@@ -325,15 +338,29 @@ public static DateTime DairyLastTradeDate(DateTime time, TimeSpan? lastTradeTime
         /// Returns the number of months prior to the contract month that the future expires
         /// </summary>
         /// <param name="underlying">The future symbol</param>
+        /// <param name="expiryDate">Expiry date to use to look up contract month delta</param>
         /// <returns>The number of months prior it expires</returns>
-        public static int ExpiresInPreviousMonth(string underlying)
+        public static int ExpiresInPreviousMonth(string underlying, DateTime? expiryDate = null)
         {
             int value;
+            if (expiryDate != null && _dairyUnderlying.Contains(underlying))
+            {
+                var dairyReportDate = expiryDate.Value.Date.AddDays(1);
+                if (_reverseDairyReportDates.ContainsKey(dairyReportDate))
+                {
+                    var contractMonth = _reverseDairyReportDates[dairyReportDate];
+                    // Gets the distance between two months in months
+                    return ((contractMonth.Year - dairyReportDate.Year) * 12) + contractMonth.Month - dairyReportDate.Month;
```

Comment:
Cool! So due to `Dairy can have an expiry after the contract month` this could return a negative value? Maybe we should rename this function so it's more precise

> DateTime? expiryDate = null

We need this argument because dairy isn't consistent?

Reply:
I second the rename, kind of throws me off on in FutureSymbols. If I am not mistaken this is to calculate how many months in the past the expiry is from the contract date?

Reply:
Yes, will rename 👍 

And yes, it's to calculate the delta between the expiry and the contract month (positive if contract expires in month before contract month, negative if expiry is in month after contract month).